reflex/vars.py

```diff
@@ -843,7 +852,17 @@ def length(self) -> Var:
             _var_is_string=False,
         )
 
-    def __eq__(self, other: Var) -> Var:
+    def type(self) -> Var:
```

Comment:
```suggestion
    def _type(self) -> Var:
```

Recommend using an under prefix name for any non-operator function defined in the `Var` class to avoid shadowing potential field names in objects represented by the Var that are accessed via `__getattr__`. This is why `name` and friends were renamed to `_var_name`

```python
class Foo(rx.Base):
    name: str = ""
    type: str = ""


class State(rx.State):
    f: Foo = Foo()

...

rx.cond(State.f.type == "pasta", ...)
```

☝️ this (potentially existing) code will break with this change

---------------

yes, shadowing underscore names is also not ideal, but these are less likely to be used in user-facing code.

Reply:
> Recommend using an under prefix name for any non-operator function defined in the `Var` class to avoid shadowing potential field names in objects represented by the Var that are accessed via `__getattr__`. This is why `name` and friends were renamed to `_var_name`
> 
> ```python
> class Foo(rx.Base):
>     name: str = ""
>     type: str = ""
> 
> 
> class State(rx.State):
>     f: Foo = Foo()
> 
> ...
> 
> rx.cond(State.f.type == "pasta", ...)
> ```
> 
> ☝️ this (potentially existing) code will break with this change
> 
> yes, shadowing underscore names is also not ideal, but these are less likely to be used in user-facing code.

got it, What does that mean for existing methods like `reverse`, `length`, etc though? Do those need to converted to prefixed-underscores as well?

Reply:
> What does that mean for existing methods like `reverse`, `length`, etc though?

They probably should be renamed, but we haven't really gotten any complaints about shadowing those yet and they're already in the public API, so a frivolous rename would just break code.

I _do_ think before 1.0, we need a general solution to this problem, because the underscore names don't look great in IDE (like pycharm underlines them super ugly), and people might not think they can use the methods safely due to the underscore prefix.

I don't know what that solution looks like yet.

For now, lets just try to avoid shadowing additional names and plan to sort it out after the radix dust settles.